console/src/components/Services/Events/Common/Components/EventsSubTable.tsx

```diff
@@ -1,85 +1,183 @@
+          SubComponent={(logRow: any) => {
+            const invocationLog = rows[logRow.index];
+            const currentPayload = JSON.stringify(
+              invocationLog.request,
+              null,
+              4
+            );
+            const finalResponse = JSON.stringify(
+              invocationLog.response,
+              null,
+              4
+            );
+            return (
+              <InvocationLogDetails
+                requestPayload={currentPayload}
+                responsePayload={finalResponse}
+              />
+            );
+          }}
+        />
+      ) : (
+        <div>No data available</div>
+      )}
     </div>
+    <br />
+    <br />
+  </div>
+);
+
+const EventsSubTable: React.FC<Props> = ({
+  makeAPICall,
+  triggerType,
+  ...props
+}) => {
+  const [inv, setInvocations] = React.useState([]);
+  const [errInfo, setErrInfo] = React.useState(null);
+
+  React.useEffect(() => {
+    if (!triggerType || !props.event.id) {
+      return;
+    }
+    // TODO: handle a "loading" state
+    props
+      .invocationsData(triggerType, props.event.id)
+      .then(data => {
+        if (data && data?.invocations && !data.error) {
+          setInvocations(data.invocations);
+          return;
+        }
+        setErrInfo(data.error);
+      })
+      .catch(err => setErrInfo(err));
+  }, []);
+
+  if (!makeAPICall || !triggerType) {
+    return (
+      <RenderEventSubTable
+        event={props.event}
+        rowsFormatted={props.rowsFormatted}
+        headings={props.headings}
+        rows={props.rows}
+      />
+    );
+  }
+
+  if (errInfo) {
+    return (
+      <div>
+        <br />
+        Error occurred in fetching information about recent invocations.
+        <br />
+      </div>
+    );
+  }
+
+  const invocationRows = inv.map((r: any, i: number) => {
+    const newRow: Record<string, JSX.Element> = {};
+    // Insert cells corresponding to all rows
+    invocationColumns.forEach(col => {
+      newRow[col] = (
+        <div
+          className={styles.tableCellCenterAlignedOverflow}
+          key={`${col}-${col}-${i}`}
+        >
+          {sanitiseRow(col, r)}
+        </div>
+      );
+    });
+    return newRow;
+  });
+
+  return (
+    <RenderEventSubTable
+      event={props.event}
+      rows={inv}
+      rowsFormatted={invocationRows}
+      headings={props.headings}
+    />
   );
 };
 
-export default EventsSubTable;
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  invocationsData: (triggerType: SupportedEvents, event_id: string) =>
+    dispatch(getEventInvocationThunk(triggerType, event_id))
+      .then(data => ({
+        invocations: data.invocations,
+        error: null,
+      }))
+      .catch(err => ({
+        invocations: [],
+        error: err,
+      })),
+});
+const connector = connect(null, mapDispatchToProps);
+type InjectedReduxProps = ConnectedProps<typeof connector>;
+const connectedEventSubTable = connector(EventsSubTable);
```

Comment:
```suggestion
const ConnectedEventSubTable = connector(EventsSubTable);
```

This is a component.